### PR TITLE
Exercise 5: Immutable Infrastructure

### PR DESCRIPTION
**Pull request by robin-leach**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM python:3.8.5-buster as base
+COPY . ./app
+WORKDIR /app
+RUN pip install "poetry==1.0.10" \
+  && poetry install
```
Comment (robin-leach) [inline Dockerfile:5]: I wasn't sure about having the `COPY` stage first,but then the `poetry install` command relies on the `pyproject.toml` file having been copied across 🤔

Comment (Jonesey13) [reply Dockerfile:5]: I'd be tempted to split this into 2 `RUN` steps with the copy command in the middle. This means you don't have to install poetry every time you update/add a dependency.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+version: "3"
```
Comment (Jonesey13) [inline docker-compose.yml:1]: I get an error with this version saying that `target` is unrecognised. Had to update this to 3.8 to get it working:
```suggestion
version: "3.8"
```

**Review by Jonesey13 — CHANGES_REQUESTED** [review]
Very nearly there! This is essentially a pass (modulo a small fix to the docker compose version).

Please apply this small change and then merge in the PR 👍

File: docker-compose.yml (hunk 2 above)
Comment (robin-leach) [reply docker-compose.yml]: Interesting - it worked for me. Updating regardless!

**Pull request merged**